src/main/sensors/gyro_filter_impl.h

```diff
@@ -27,16 +34,14 @@ static FAST_CODE void GYRO_FILTER_FUNCTION_NAME(gyroSensor_t *gyroSensor, timeDe
 #ifdef USE_GYRO_DATA_ANALYSE
         if (isDynamicFilterActive()) {
             if (gyroConfig()->dyn_fft_location == DYN_FFT_AFTER_STATIC_FILTERS) {
-                gyroDataForAnalysis = gyroADCf;
-            }
-
-            if (axis == X) {
-                GYRO_FILTER_DEBUG_SET(DEBUG_FFT, 1, lrintf(gyroDataForAnalysis));
-                GYRO_FILTER_DEBUG_SET(DEBUG_FFT_FREQ, 2, lrintf(gyroDataForAnalysis));
+                // log post-static gyro data to debug 1 and 2, since this is now the 'first tier' of filtering
+                if (axis == X) {
+                GYRO_FILTER_DEBUG_SET(DEBUG_FFT, 1, lrintf(gyroADCf));
+                GYRO_FILTER_DEBUG_SET(DEBUG_FFT_FREQ, 2, lrintf(gyroADCf));
+                }
+                gyroDataAnalysePush(&gyroSensor->gyroAnalyseState, axis, gyroADCf);
+                gyroADCf = gyroSensor->notchFilterDynApplyFn((filter_t *)&gyroSensor->notchFilterDyn[axis], gyroADCf);
             }
-
-            gyroDataAnalysePush(&gyroSensor->gyroAnalyseState, axis, gyroADCf);
```

Comment:
No need to rewrite the whole thing. The current version looks cleaner and problem is only right in this line:
gyroDataForAnalysis needs to be pushed, not gyroADCf.

Reply:
The reason for the more extensive changes is as follows:

IIn my original versions, debug 2 always showed the effect of the ‘first tier’ of filtering. 

If the user:
- set ‘dynamic first’, then debug 2 showed the effect of the dynamic,
- set ‘static first’, then debug 2 showed the effect of the statics.

So the filter cascade was always:
debug 3 = gyro unfiltered
debug 2 = gyro after first tier filtering
gyro trace itself = both

This was actually really easy for the user to interpret.  With the current code, if the user chooses pre-static, debug 2 just shows the same gyroADCf as debug 3, and doesn't show what the dynamic is actually doing.

Hence that last commit.  Without it I had people say ‘the dynamic is not working’ in pre-static mode.

Reply:
Ah, I didn't look at the first commit, now I get the reasoning. Still it looks somewhat redundant.